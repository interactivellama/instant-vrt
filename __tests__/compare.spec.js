/* global describe, beforeEach, it, xit */

const { assert } = require('chai')
const path = require('path')
const tests = require('./123.json')
const refs = require('./abc.json')

const compare = require('../compare')
const report = require('../report')

describe('compare', () => {
  it('passes', () => {
    const res = compare(tests["passing"], refs["passing"])
    assert.equal(res.passed, true)
  })

  it('puts the lines of html fail in results html', () => {
    const res = compare(tests["fail_html"], refs["fail_html"])
    assert.equal(res.passed, false)
    assert.deepEqual(res.html, [3])
  })

  it('fails pairs of the style fails in results style', () => {
    const res = compare(tests["fail_style"], refs["fail_style"])
    assert.deepEqual(res.style[0], ['border-bottom-color'])
    assert.deepEqual(res.style[3], ['display'])
  })
})

describe('report', () => {
  it('shows the html output', () => {
    const res = report(compare(tests["fail_html"], refs["fail_html"]))
    expect(res).toMatchSnapshot()
  })
  it('shows the style output', () => {
    const res = report(compare(tests["fail_style"], refs["fail_style"]))
    expect(res).toMatchSnapshot()
  })
  it('is too different', () => {
    const res = report(compare(tests["tooDifferent"], refs["tooDifferent"]))
    expect(res).toMatchSnapshot()
  })
})
