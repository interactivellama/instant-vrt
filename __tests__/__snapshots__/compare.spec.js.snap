// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`report is too different 1`] = `
Object {
  "html": "",
  "style": "33 failures
Too many failures to show",
}
`;

exports[`report shows the html output 1`] = `
Object {
  "html": "1 failures
  <header class=\\"site site-masthead slds-grid slds-wrap\\">
[31m    <div class=\\"site-masthead-title slds-has-flexi-truncate slds-align-middle\\">[39m
[32m    <div alt=\\"FAIL\\" class=\\"site-masthead-title slds-has-flexi-truncate slds-align-middle\\">[39m
      <span class=\\"slds-hide docsearch-category\\">Colors</span>",
  "style": "",
}
`;

exports[`report shows the style output 1`] = `
Object {
  "html": "",
  "style": "2 failures
<main class=\\"site-main slds-order_1\\" role=\\"main\\" id=\\"content\\">
{
border-bottom-color: [31mrgb(22, 50, 92)[39m [32mrgb(22, 666, 92),[39m
[32m[39m}
<span class=\\"slds-hide docsearch-category\\">
{
display: [31mvisible[39m [32mnone,[39m
[32m[39m}",
}
`;
